### PR TITLE
feat: add extend grant type login_queue_ticket

### PR DESCRIPTION
**Pull request by wardab0104**
feat: add extend grant type login_queue_ticket

### CODE REVIEW

File: access.go [hunk 1]
```diff
@@ -567,6 +571,55 @@ func (s *Server) handleClientCredentialsRequest(w *Response, r *http.Request) *A
...
+		}
+		client = getClient(auth, w.Storage, w)
+	}
+
+	// generate access token
+	ret := &AccessRequest{
+		Type:            EXTEND_LOGIN_QUEUE_TICKET,
+		Code:            r.Form.Get("login_queue_ticket"),
+		CodeVerifier:    r.Form.Get("code_verifier"),
+		GenerateRefresh: true,
```
Comment (Allen8975701) [inline access.go:605]: do we need refresh token here?

File: access.go [hunk 2]
```diff
@@ -567,6 +571,55 @@ func (s *Server) handleClientCredentialsRequest(w *Response, r *http.Request) *A
...
+			return nil
+		}
+		client = getClient(auth, w.Storage, w)
+	}
+
+	// generate access token
+	ret := &AccessRequest{
+		Type:            EXTEND_LOGIN_QUEUE_TICKET,
+		Code:            r.Form.Get("login_queue_ticket"),
+		CodeVerifier:    r.Form.Get("code_verifier"),
```
Comment (Allen8975701) [inline access.go:604]: will it have code_verify here?

File: access.go (hunk 1 above)
Comment (wardab0104) [reply access.go:605]: yes, this is for user token

File: access.go (hunk 2 above)
Comment (wardab0104) [reply access.go:604]: I just keep the param here in case we need to implement the code challenge & code verifier validation in login queue service

**Comment by christianrahardja-ab** [comment]
looks good to me

**Review by angelo-chan — APPROVED** [review]

**Review by marselsampe — APPROVED** [review]

**Pull request merged**